### ISSUE TITLE
ActLike() fails with explicit interface implementation

### ISSUE THREAD

**Issue by piedar** [ekonbenefits/impromptu-interface#36]
I'm not sure if this is expected behavior.

```csharp
interface IDo
{
    void Do();
}

sealed class DoNothing : IDo
{
    void IDo.Do() { }
}
```

```csharp
class TheDoNothing
{
    [Test]
    public void CanActLike()
    {
        var instance = new DoNothing();
        var acting = instance.ActLike<IDo>();
        acting.Do(); // throws RuntimeBinderException
    }
}
```

I read about [Dynamitey UsagePrivate](https://github.com/ekonbenefits/dynamitey/wiki/UsagePrivate) but this produces the same result.

```csharp
class TheDoNothing
{
    [Test]
    public void CanActLike()
    {
        var instance = new DoNothing();
        var context = new InvokeContext(instance, typeof(IDo));
        var acting = context.ActLike<IDo>();
        acting.Do(); // throws RuntimeBinderException
    }
}
```

Obviously this a toy example, but I'm running into real issues when trying to do something similar using DynamicActLike() with an interface type known only at runtime.

**Comment by jbtule** [comment]
Yeah, with this toy example, I'd just test to see if instance already implements IDo and not run ActLike. 

If it's not the same interface, explicit interfaces are very tricky, because the methods names are all qualified. How do I know what the name of the interface you are intending, I'd have do an exhaustive search with reflection,

**Comment by piedar** [comment]
My use case is to make an instance loaded from one `AssemblyLoadContext` act like an interface loaded from a different `AssemblyLoadContext`.  It is the same interface by name, but technically a different `Type` because it comes from a different load context.  This looks something like

```csharp
// find a matching Type from the assemblies loaded with the Provider's AssemblyLoadContext
Type resolvedType = ResolveType(serviceType);
object resolvedService = Provider.GetService(resolvedType);
var result = ImpromptuInterface.Impromptu.DynamicActLike(resolvedService, serviceType);
```

This almost works great, except that when the serviceType is `Microsoft.Extensions.Logging.Abstractions.ILogger<T>` the Provider produces a `Logger<T>` which has explicit implementations of the interface methods.

**Comment by jbtule** [comment]
Confirmed on master (net8.0), both spellings:

```
instance.ActLike<IDo>().Do()                         !! RuntimeBinderException: 'DoNothing' does not contain a definition for 'Do'
new InvokeContext(instance, typeof(IDo)).ActLike<IDo>().Do()   !! same
```

This is the C# runtime binder's rule rather than a bug in the emit: an explicit interface implementation compiles to a private method named `IDo.Do`, and the binder only sees the runtime type's ordinary members. `UsagePrivate` does not help, because the problem is not accessibility — the member simply is not there under that name. The same restriction applies to C# `dynamic` itself.

The practical fix for this shape is #27: if the target already implements the interface, the proxy has nothing to add and could return the instance (or call through the interface) instead of binding dynamically by name. That would make this case work exactly, and it is also faster. Linking the two.
